addons/mail/models/mail_message.py

```diff
@@ -988,11 +999,17 @@ def _message_format(self, fnames):
                         'field_type': tracking.field_type,
                     })
 
+            if message_sudo.model and message_sudo.res_id:
```

Comment:
Shouldn't you try to group by model in order to fetch all names at once ?

Reply:
@tde-banana-odoo The first implementation did that. But @seb-odoo noted that it was already done internally by the ORM. We could use the ORM cache with `browse` and let it do the heavy work.
As I don't fully understand how all this works, my explanation could be a bit confusing.

Reply:
Yep, this is done just above with `threads.sudo().mapped('display_name')` which is doing one query per model and filling the cache.

If this is not clear enough to do it that way, this loop above can be replaced by changing the line below here by making explicit what to prefetch with `with_prefetch`.

`record_name = self.env[message_sudo.model].browse(message_sudo.res_id).with_prefetch(threads_by_model_name[message_sudo.model].ids).display_name`

But while this make the prefetch itself explicit, I don't really find the code easier to read with this change. Maybe instead if really needed to avoid confusion just add a comment above the browse to note that the data is indeed already prefetched before the loop?

Reply:
Well it is a bit magic as it depends on code just above. We could easily break the prefetch optimization when splitting / working on this method without noticing it.

Having a with_prefetch makes it explicit which is always better when dealing with that kind of ORM stuff IMHO. If I modify the line, I know there is "something to check with prefetching". Otherwise I may miss it and add potential performance issues.

Reply:
Sounds fair, I'm fine either way, so let's go for the more explicit prefetching here.

Reply:
Note that in this case `threads_by_model_name` can probably become `thread_ids_by_model_name` and this has the added benefit of getting rid of the other browse.

Reply:
@sea-odoo I have edited the PR.
@tde-banana-odoo Thank's for you input, I think it's a lot better now.

Reply:
Side remark: could you maybe just write the line on a multi line :D ? My editor keeps scrolling which indicates it becomes quite long. We don't enforce line length in python but about 100~120 characters time has come to multi-line :) .

Reply:
done !